src/to_wkt/geo_trait_impl.rs

```diff
@@ -0,0 +1,446 @@
+use core::fmt;
+use std::fmt::Write;
+
+use geo_traits::{
+    CoordTrait, GeometryCollectionTrait, GeometryTrait, LineStringTrait, LineTrait,
+    MultiLineStringTrait, MultiPointTrait, MultiPolygonTrait, PointTrait, PolygonTrait, RectTrait,
+    TriangleTrait,
+};
+use geo_types::CoordNum;
+
+use crate::types::{Coord, LineString, Polygon};
+use crate::WktNum;
+
+/// The physical size of the coordinate dimension
+///
+/// This is used so that we don't have to call `.dim()` on **every** coordinate. We infer it once
+/// from the `geo_traits::Dimensions` and then pass it to each coordinate.
+#[derive(Clone, Copy)]
+pub(crate) enum PhysicalCoordinateDimension {
+    Two,
+    Three,
+    Four,
+}
+
+impl From<geo_traits::Dimensions> for PhysicalCoordinateDimension {
+    fn from(value: geo_traits::Dimensions) -> Self {
+        match value.size() {
+            2 => Self::Two,
+            3 => Self::Three,
+            4 => Self::Four,
+            size => panic!("Unexpected dimension for coordinate: {}", size),
+        }
+    }
+}
+
+pub fn point_to_wkt<T: CoordNum + WktNum + fmt::Display, G: PointTrait<T = T>, W: Write>(
```

Comment:
naming nit for all the `foo_to_wkt` methods: Idiomatically, I'd expect `point_to_wkt` to do something like `From::from(point) -> Wkt`.

But this doesn't convert to Wkt. Internally it conceptually converts to something like a String of wkt, but we don't return that either.

A closer analogy might be the `Format::write` method or `write` macro, implying something like:
```
write_wkt_point(&mut writer, &my_point)?;
write_wkt_polygon(&mut writer, &my_polygon)?;
```

In fact, I think we can exclude `wkt` from the method names since it's redundant with the crate.

```
write_point(&mut writer, &my_point)?;
write_polygon(&mut writer, &my_polygon)?;
```

If calling users want to be explicit about the fact that it's *wkt* at the callsite, they could:
```
wkt::write_point(&mut writer, &my_point)?;
wkt::write_polygon(&mut writer, &my_point)?;
```

Note: this would entail re-exporting all these methods at the top-level of the crate.


Additionally, taking advantage of https://github.com/georust/geo/pull/1157#issuecomment-2408643696 maybe there should be a plain old:

```
fn write(&mut writer, any_geometry: impl GeometryTrait) {
   write_geometry(writer, any_geometry)
}
```

So the caller can do:
```
wkt::write(&mut writer, &my_point);
wkt::write(&mut writer, &my_polygon);
```

I think that'd work, right? It'd be slightly less performant if you already know the type, but probably negligible for most, and nicer to write.

Reply:
Yes, this is a good point. I had the same realization when cleaning up my `wkb` crate: https://github.com/kylebarron/wkb/pull/4

Reply:
I shortened names but didn't shorten `write_geometry` to `write`. We should be able to just `pub use write_geometry as write`?